Feat(clouddev-2246): new resource edgecenter_mkaas

Uiopas opened the pull request:
- new resource edgecenter_mkaas_cluster
- new resource edgecenter_mkaas_pool

Uiopas committed:
feat(CLOUDDEV-2246): new mkaas_cluster and mkaas_pool resources

Leonid-St commented on edgecenter/resource_edgecenter_mkaas_pool.go:
```diff
@@ -0,0 +1,351 @@
+		return diag.FromErr(err)
+	}
+
+	poolID := taskResult.MkaasPools[0]
```
Тоже самое что и выше - проверку на нил и первый элемент.

Leonid-St commented on edgecenter/resource_edgecenter_mkaas_cluster.go:
```diff
@@ -0,0 +1,359 @@
+	}
+
+	clusterID := taskResult.MkaasClusters[0]
+	log.Printf("[DEBUG] MKaaS id (from taskResult): %.0f", clusterID)
```
здесь почему то и везде в коде в  логе дебаг остался.

Leonid-St commented on edgecenter/resource_edgecenter_mkaas_cluster.go:
```diff
@@ -0,0 +1,359 @@
+
+	clusterID := taskResult.MkaasClusters[0]
+	log.Printf("[DEBUG] MKaaS id (from taskResult): %.0f", clusterID)
+	d.SetId(strconv.FormatFloat(clusterID, 'f', -1, 64))
```
а почему кластерID типа float ?

Leonid-St commented on edgecenter/resource_edgecenter_mkaas_cluster.go:
```diff
@@ -0,0 +1,359 @@
+	clusterID := taskResult.MkaasClusters[0]
+	log.Printf("[DEBUG] MKaaS id (from taskResult): %.0f", clusterID)
+	d.SetId(strconv.FormatFloat(clusterID, 'f', -1, 64))
+	resourceMkaasClusterRead(ctx, d, m)
```
эта функция разе ничего не возвращает ? Вроде она diags как раз возвращает ?

Leonid-St commented on edgecenter/resource_edgecenter_mkaas_cluster.go:
```diff
@@ -0,0 +1,359 @@
+
+	cluster, resp, err := clientV2.MkaaS.ClusterGet(ctx, clusterID)
+	if err != nil {
+		if resp.StatusCode == http.StatusNotFound {
```
Вот это хорошо. Проверь по коду чтобы везде была это  обработка статуса not found где вызывается clientV2.МETHOD

DamirZinatullin reviewed (commented):
круто, работает)

dodgyturtle commented on edgecenter/resource_edgecenter_mkaas_cluster.go:
```diff
@@ -0,0 +1,360 @@
+		},
+		Importer: &schema.ResourceImporter{
+			StateContext: func(ctx context.Context, d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+				projectID, regionID, k8sID, err := ImportStringParser(d.Id())
```
k8sID это MKaaSClusterID?

dodgyturtle commented on edgecenter/resource_edgecenter_mkaas_cluster.go:
```diff
@@ -0,0 +1,360 @@
+	MKaaSClusterCreateTimeout    = 30 * time.Minute
+	MKaaSClusterUpdateTimeout    = 30 * time.Minute
+	MKaaSClusterDeleteTimeout    = 20 * time.Minute
+	MKaaSClusterKeypairNameField = "ssh_keypair_name"
```
Название для полей лежат в `edgecenter/provider.go`. Лучше их сделать максимально унифицированными. Без приставки MKaaS. Чтобы можно было переиспользовать в дальнейшем.

dodgyturtle replied:
Это касается всех, что написаны ниже

Leonid-St replied:
здесь решили не менять

DamirZinatullin commented on edgecenter/resource_edgecenter_mkaas_cluster.go:
```diff
@@ -0,0 +1,360 @@
+		ReadContext:   resourceMKaaSClusterRead,
+		UpdateContext: resourceMKaaSClusterUpdate,
+		DeleteContext: resourceMKaaSClusterDelete,
+		Description:   "Represent k8s cluster.",
```
ну по смыслу у нас mkaas - это сервис, а сам кластер это k8s

dodgyturtle replied:
Тогда может Kubernetes?

dodgyturtle replied:
Опять же: resourceMKaaSCluster

dodgyturtle commented on edgecenter/resource_edgecenter_mkaas_cluster.go:
```diff
@@ -0,0 +1,360 @@
+
+				opts := edgecloudV2.MkaaSClusterUpdateRequest{
+					Name:            d.Get(NameField).(string),
+					MasterNodeCount: nodeCount,
```
Т.е. ты будешь update делать со старыми данными nodeCount, если у тебя изменилось имя? Eco нормально это отработает?

Leonid-St commented on edgecenter/resource_edgecenter_mkaas_cluster.go:
```diff
@@ -0,0 +1,360 @@
+				if err != nil {
+					return nil, err
+				}
+				d.Set("project_id", projectID)
```
эту же позицию год назад продвигал когда тф занимался. дамир тогда мне сказал что такое делать не надо)

DamirZinatullin commented on edgecenter/resource_edgecenter_mkaas_cluster.go:
```diff
@@ -0,0 +1,367 @@
+		},
+		Importer: &schema.ResourceImporter{
+			StateContext: func(ctx context.Context, d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+				projectID, MKaaSClusterID, k8sID, err := ImportStringParser(d.Id())
```
почему MKaaSClusterID???

DamirZinatullin commented on edgecenter/resource_edgecenter_mkaas_pool.go:
```diff
@@ -0,0 +1,344 @@
+}
+
+func resourceMKaaSPoolCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	log.Println("Start MKaaS Cluster creating")
```
почему log. оставил?

Leonid-St commented on edgecenter/resource_edgecenter_mkaas_cluster.go:
```diff
@@ -0,0 +1,360 @@
+
+func resourceMKaaSClusterRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	log.Println("Start MKaaS reading")
+	var diags diag.Diagnostics
```
можно делать return diag.Diagnostics{}. Но  как будто по коду у нас везде принято создавать эту переменную вначале

Leonid-St commented on edgecenter/resource_edgecenter_mkaas_cluster.go:
```diff
@@ -0,0 +1,367 @@
+					opts.Name = NameField
+				}
+
+				if d.HasChange(MKaaSClusterControlPlaneField) {
```
поменял на d.HasChange(MKaaSClusterNodeCountField)

DamirZinatullin committed:
feat(CLOUDDEV-2379): added publish_kube_api_to_internet attribute to mkaas_cluster resource (#255)

Co-authored-by: damir.zinatullin <damir.zinatullin@edgecenter.ru>

Uiopas committed:
feat(CLOUDDEV-2245): e2e tests (#253)

DamirZinatullin committed:
feat(CLOUDDEV-2245): change ApplyE to ApplyAndIndempotentE (#257)

Co-authored-by: damir.zinatullin <damir.zinatullin@edgecenter.ru>

Leonid-St committed:
feat(CLOUDDEV-2454): mkaas: cluster pool data source

* feat(CLOUDDEV-2454): mkaas: data source cluster pool

* feat(CLOUDDEV-2454): go version ->1.25.4 in go.mod, readme, .github

---------

Co-authored-by: Leonid Stepanov <leonid.stepanov@edgecener.ru>

GeorgeD7857 committed:
feat(CLOUDDEV-2465): added pod_subnet and service_subnet to clusterResource (#259)

GeorgeD7857 committed:
feat(CLOUDDEV-2451-add-datasource-cluster): add datasource cluster (#265)

Zwergpro committed:
feat(CLOUDDEV-2664): Refactor MKaaS pool to support multiple security group IDs

GeorgeD7857 committed:
feat(feat/CLOUDDEV-2629-update-pool): change mkaas update pool (#268)

Zwergpro committed:
refactor(CLOUDDEV-2650): Rename "state" field to "stage" across documentation, code, and tests (#269)

GeorgeD7857 committed:
feat(CLOUDDEV-2542-error-unsupported-fields): add errors for update unsupported fields mkaas (#270)

Leonid-St committed:
feat(CLOUDMKAAS-479): update sg in pool

feat(CLOUDMKAAS-479): update sg in pool + golangci-lint run --fix + go1.24.4

---------

Co-authored-by: Leonid Stepanov <leonid.stepanov@edgecener.ru>

Leonid-St committed:
feat(CLOUDMKAAS-815): single params in update cluster (#276)

* feat(CLOUDMKAAS-815): single params in update cluster

---------

Co-authored-by: Leonid Stepanov <leonid.stepanov@edgecener.ru>

Zwergpro committed:
feat(CLOUDMKAAS-805): Bump edgecentercloud-go (#281)

Zwergpro committed:
feat(CLOUDMKAAS-830): Split MKaaS cluster updates into separate requests for name and master node count (#277)

Leonid-St committed:
feat(CLOUDMKAAS-825): support adding pool labels (#280)

* feat(CLOUDMKAAS-825): support adding pool labels

---------

Co-authored-by: Leonid Stepanov <leonid.stepanov@edgecener.ru>

Zwergpro committed:
Feat/cloudmkaas 868 update lables (#286)

GeorgeD7857 committed:
feat(CLOUDMKAAS-985): Increase minimum control plane volume size to 30 GiB (#304)
